src/app/components/wrapper-components/part-creation/part-creation.component.spec.ts

```diff
@@ -148,7 +148,7 @@ describe('PartCreationComponent', () => {
                 await mockCandidateArrival();
 
                 // Then it is possible to choose a candidate
-                expect(component.currentJoiner).toEqual(JoinerMocks.WITH_FIRST_CANDIDATE);
+                expect(component.currentJoiner).toEqual(JoinerMocks.INITIAL);
```

Comment:
ça fait vachement moins logique à lire comme ça avec la sub-collection :|

Reply:
bah, "it is possible to choose a candidate" se traduit en code en "je clique sur le choix d'opposant et ça marche", assez logique non ? Après on pour remplacer le check sur currentJoiner par un check sur la liste des candidats en interne du composant, mais ça me semble trop précis de tester l'état interne du composant non ?

Reply:
le Then d'avant devrais avoir été "Then currentJoiner now include a MinimalUser in candidate field"

Là on test que c'est toujours initial :p Est ce que commenter le when casse le test seulement :p !

Commenter le when doit niquer tout test, normalement, sinon il teste rien, sauf évidemment pour "when doing this then nothing should have changed" qui peut être un test souhaitable par-ci par-là.

C'était de ce genre de test que je parlais genre, expect(component.subCollectionOfUser).toHaveElement(newGuy)

Reply:
bah oui commenter le when casse le test, vu que `#chooseOpponent` n'existe pas alors ! On teste le composant, donc on s'assure que quand un candidat arrive, on le voit sur le composant. Qu'il soit présent dans le joiner ou dans une sous-collection, c'est pas la responsabilité du composant.

Reply:
Mais la première ligne du then ne serait pas cassée alors qu'elle était cassée avant.
Ce serait bien d'avoir un "expectSubCollectionToBe([JOSER, ANTOINE]) !
Car là on test plus le joiner en lui même mais ça subcollection de candidats.

Reply:
mais ce qui compte c'est que le test casse non ? Puis on veut garder la première ligne pour s'assurer qu'il n'y ait pas de changements louches en plus. Et ça me semble violer le principe de séparation de tester des choses qui ne sont pas de l'ordre du composant ? Si hypothétiquement un jour on passe pas par firebase, on veut pouvoir garder le code des composants & leurs tests sans devoir tout adapter ? Et JoinerDAO s'assure que quand addCandidate est appelé, la subCollection est bien mise à jour.

Reply:
Ouais, belle séparation des concernes ! Tu peux résoudre